names/names.py

```diff
@@ -10,13 +10,32 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
-duplicates = []  # Return the list of duplicates in this data structure
+# *
+# * Start solutions
+# *
 
-# Replace the nested for loops below with your improvements
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+# duplicates = []  # Return the list of duplicates in this data structure
+
+
+# Fastest solution: (STRETCH SOLUTION)
+duplicates = list(set(names_1).intersection(names_2))
```

Comment:
This is perfect ✨ 

Sets are absolutely the right call here. The only question I have is this: Is it required that you add another iteration + a conversion by placing it in a list? Would it not also pass the test if it was:

```python
duplicates = set(names_1).intersection(names_2)
```

The above would return a set of names containing no duplicates, and give you `O(k + n)` time (since it has to iterate both lists to get the intersection between both lists).

Give it a shot and see just how big a difference in time it saves your code.

Reply:
I know, would look much cleaner 😞 the tests definitely are looking for a list here. Thanks for checking this out!